tests/integration_tests/modules/test_power_state_change.py

```diff
@@ -0,0 +1,79 @@
+"""Integration test of the cc_power_state_change module.
+
+Test that the power state config options work as expected.
+"""
+
+from tests.integration_tests.log_utils import ordered_items_in_text
+import pytest
+import time
+
+from tests.integration_tests.clouds import IntegrationCloud
+from tests.integration_tests.instances import IntegrationInstance
+
+USER_DATA = """\
+#cloud-config
+power_state:
+  delay: {delay}
+  mode: {mode}
+  message: msg
+  timeout: {timeout}
+  condition: {condition}
+"""
+
+
+def _detect_reboot(instance: IntegrationInstance):
+    # We'll wait for instance up here, but we don't know if we're
+    # detecting the first boot or second boot, so we also check
+    # the logs to ensure we've booted twice. If the logs show we've
+    # only booted once, wait until we've booted twice
+    instance.instance.wait()
+    for _ in range(600):
+        try:
+            log = instance.read_from_file('/var/log/cloud-init.log')
```

Comment:
I agree for a general implementation, but in this specific case, the test code isn't initiating the reboot. Since we don't clear the reboot count when we generate the snapshot image, we can't definitively know how many reboots we should be looking for.

Reply:
Very good point! You _could_ do something with `bootcmd` to determine it, but it would still be test-specific so there's really no reason to think it would be better.